types/localizejs/index.d.ts

```diff
@@ -0,0 +1,225 @@
+         * in the URL is used to check to detect the language, ie. www.localize.com/fr. If no language dictionary exists for that
+         * segment then the language setting in their browser is used.
+         */
+        autodetectLanguage: boolean;
+
+        /**
+         * Defaults to false. When true, Localize will attempt to translate the entire body of the page.
+         * If false, Localize will only translate content contained with a "localizejs" class name.
+         */
+        translateBody: boolean;
+
+        /**
+         * The default language your website will be in when no language has been selected. Defaults to the source language of your website.
+         */
+        defaultLanguage: string;
+
+        /**
+         * The base path will be stripped from the URL of the phrase as seen in the "Filter by pages" feature.
+         */
+        basePath: string;
+
+        /**
+         * Defaults to true. If true, the <title> of the page will translate.
+         */
+        translateTitle: boolean;
+
+        /**
+         * Defaults to false. Allows users to turn on meta tag translation. This optimizes your site for SEO.
+         */
+        translateMetaTags: boolean;
+
+        /**
+         * Defaults to false. If true, Localize will detect phrases only when the page is not translated.
+         * Please contact support@localizejs.com prior to updating this option.
+         */
+        saveNewPhrasesFromSource: boolean;
+
+        /**
+         * Defaults to false. Automatically translate content that is added dynamically to your webpage.
+         * For example, if your webpage dynamically adds html into the source of the page, our library
+         * will translate it once the translations have been generated. Behind the scenes this means the
+         * dictionary file with all your translated content is available for use with Localize.translate().
+         * However, translations are not generated instantly, so use with our library event updatedDictionary is recommended.
+         */
+        retranslateOnNewPhrases: boolean;
+
+        /**
+         * Defaults to false. If true, the Localize library will not send additional metadata to our servers.
+         * This metadata includes the surrounding HTML of the phrases detected on your website.
+         */
+        enhancedContentSecurity: boolean;
+
+        /**
+         * Defaults to false. If true, the Localize library will pick up phrases in the <time> elements.
+         */
+        translateTimeElement: boolean;
+
+        /**
+         * Defaults to false. If true, the Localize library will pick up numbers as phrases.
+         */
+        translateNumbers: boolean;
+    }
+}
+
+declare var Localize: {
+    /**
+     * Initializes LocalizeJS with the supplied options.
+     * @param options  An object containing the supplied options.
+     */
+    initialize(options: LocalizeJS.Context.Options): void;
```

Comment:
I didn't find any API related to the `initialize` at https://www.npmjs.com/package/localizejs , is it same package?

Reply:
No this is not the same package, sorry for the confusion. I tried to outline this in the header of the request. The end service is the same however the NPM package appears to be specific to their node implementation however we need this for an application using the web browser snippet (similar to how Google Analytics is added). Should I use a different name to avoid the confusion and if so do you have a recommendation?

Reply:
I'm not sure, let's ask @andy-ms

Reply:
Just to point out, the package on Github point out it is deprecated and defer to the web API:

```
This module has been deprecated

Please use the official Localize API.
```

Since they have no other library other than the one offered by the web snippet do you think it is safe continuing to use this name? The version on the current package is 0.4.2 and I have used 1.0 here which is greater.

Do you think knowing this helps? I am needing this for a current project and would like to stop using the local copy. I am correct in assuming the type can't be published to npm before merging into this repository?

Thanks!

Reply:
I'm not sure, according to the document
>  If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.

You may have to change the name to avoid conflict with existing package.
This is a weekend, so it seems no mergers online now, so you may have to wait a few days.

@alloy @johnnyreilly Can you give suggestions for the no npm package case?

Reply:
Sorry, this is out of my wheelhouse.

Reply:
I noticed they called this the Library API throughout their documentation and links so I have renamed this to `LocalizeJS Library` and `localizejs-library` which no package exists for. Let me know if this helps.

Thank you!

Reply:
Yes, use another name -- `localizejs-library` is good.